src/geometry.F90

```diff
@@ -130,6 +125,7 @@ recursive subroutine find_cell(p, found, search_cells)
     logical,        intent(inout) :: found
     integer,        optional      :: search_cells(:)
     integer :: i                    ! index over cells
+    integer :: j
```

Comment:
perhaps ``n_coord`` or ``coord_num`` would be a more transparent var name for the new "j" local vars in a lot of these routines?